Flaky-test: NamespacesTest.testSetOffloadThreshold

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/3530823792/jobs/5923383857

### Exception stacktrace

```
2022-11-23T10:06:21.6045306Z [ERROR] Tests run: 96, Failures: 1, Errors: 0, Skipped: 70, Time elapsed: 80.822 s <<< FAILURE! - in org.apache.pulsar.broker.admin.NamespacesTest
2022-11-23T10:06:21.6049385Z [ERROR] testSetOffloadThreshold(org.apache.pulsar.broker.admin.NamespacesTest)  Time elapsed: 0.253 s  <<< FAILURE!
2022-11-23T10:06:21.6050494Z java.lang.AssertionError: expected [100] but found [-1]
2022-11-23T10:06:21.6051266Z 	at org.testng.Assert.fail(Assert.java:110)
2022-11-23T10:06:21.6051687Z 	at org.testng.Assert.failNotEquals(Assert.java:1413)
2022-11-23T10:06:21.6052086Z 	at org.testng.Assert.assertEqualsImpl(Assert.java:149)
2022-11-23T10:06:21.6161964Z 	at org.testng.Assert.assertEquals(Assert.java:131)
2022-11-23T10:06:21.6164814Z 	at org.testng.Assert.assertEquals(Assert.java:911)
2022-11-23T10:06:21.6170616Z 	at org.testng.Assert.assertEquals(Assert.java:945)
2022-11-23T10:06:21.6173421Z 	at org.apache.pulsar.broker.admin.NamespacesTest.testSetOffloadThreshold(NamespacesTest.java:1496)
2022-11-23T10:06:21.6176759Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-11-23T10:06:21.6179568Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-11-23T10:06:21.6182447Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-11-23T10:06:21.6185918Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-11-23T10:06:21.6189100Z 	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
2022-11-23T10:06:21.6192284Z 	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
2022-11-23T10:06:21.6195463Z 	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
2022-11-23T10:06:21.6198382Z 	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
2022-11-23T10:06:21.6201047Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-11-23T10:06:21.6203701Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-11-23T10:06:21.6206392Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-11-23T10:06:21.6208964Z 	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/NamespacesTest.java
@@ -1457,46 +1457,25 @@ public class NamespacesTest extends MockedPulsarServiceBaseTest {
         TopicName topicName = TopicName.get("persistent", this.testTenant, "offload", "offload-topic");
         String namespace =  topicName.getNamespaceObject().toString();
         System.out.println(namespace);
-        // set a default
-        pulsar.getConfiguration().setManagedLedgerOffloadAutoTriggerSizeThresholdBytes(1);
-        pulsar.getConfiguration().setManagedLedgerOffloadThresholdInSeconds(100);
         // create the namespace
         admin.namespaces().createNamespace(namespace, Set.of(testLocalCluster));
         admin.topics().createNonPartitionedTopic(topicName.toString());
 
         admin.namespaces().setOffloadDeleteLag(namespace, 10000, TimeUnit.SECONDS);
+        // assert we get -1 which indicates it will fall back to default
         assertEquals(admin.namespaces().getOffloadThreshold(namespace), -1);
         assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), -1);
-
-        // assert we get the default which indicates it will fall back to default
-        assertEquals(admin.namespaces().getOffloadThreshold(namespace), -1);
-        assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), -1);
-        // the ledger config should have the expected value
-        ManagedLedgerConfig ledgerConf = pulsar.getBrokerService().getManagedLedgerConfig(topicName).get();
-        MockLedgerOffloader offloader = new MockLedgerOffloader(OffloadPoliciesImpl.create("S3", "", "", "",
-                null, null,
-                null, null,
-                OffloadPoliciesImpl.DEFAULT_MAX_BLOCK_SIZE_IN_BYTES,
-                OffloadPoliciesImpl.DEFAULT_READ_BUFFER_SIZE_IN_BYTES,
-                admin.namespaces().getOffloadThreshold(namespace),
-                admin.namespaces().getOffloadThresholdInSeconds(namespace),
-                pulsar.getConfiguration().getManagedLedgerOffloadDeletionLagMs(),
-                OffloadPoliciesImpl.DEFAULT_OFFLOADED_READ_PRIORITY));
-        ledgerConf.setLedgerOffloader(offloader);
-        assertEquals(ledgerConf.getLedgerOffloader().getOffloadPolicies().getManagedLedgerOffloadThresholdInBytes(),
-                Long.valueOf(-1));
-        assertEquals(ledgerConf.getLedgerOffloader().getOffloadPolicies().getManagedLedgerOffloadThresholdInSeconds(),
-                Long.valueOf(-1));
-
 
         // set an override for the namespace
         admin.namespaces().setOffloadThreshold(namespace, 100);
         admin.namespaces().setOffloadThresholdInSeconds(namespace, 100);
-        assertEquals(admin.namespaces().getOffloadThreshold(namespace), 100);
-        assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), 100);
-        ledgerConf = pulsar.getBrokerService().getManagedLedgerConfig(topicName).get();
+        Awaitility.await().untilAsserted(() -> {
+                    assertEquals(admin.namespaces().getOffloadThreshold(namespace), 100);
+                    assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), 100);
+                });
+        ManagedLedgerConfig ledgerConf = pulsar.getBrokerService().getManagedLedgerConfig(topicName).get();
         admin.namespaces().getOffloadPolicies(namespace);
-        offloader = new MockLedgerOffloader(OffloadPoliciesImpl.create("S3", "", "", "",
+        MockLedgerOffloader offloader = new MockLedgerOffloader(OffloadPoliciesImpl.create("S3", "", "", "",
                 null, null,
                 null, null,
                 OffloadPoliciesImpl.DEFAULT_MAX_BLOCK_SIZE_IN_BYTES,
@@ -1514,7 +1493,10 @@ public class NamespacesTest extends MockedPulsarServiceBaseTest {
         // set another negative value to disable
         admin.namespaces().setOffloadThreshold(namespace, -2);
         admin.namespaces().setOffloadThresholdInSeconds(namespace, -2);
-        assertEquals(-2, admin.namespaces().getOffloadThreshold(namespace));
+        Awaitility.await().untilAsserted(() -> {
+                    assertEquals(admin.namespaces().getOffloadThreshold(namespace), -2);
+                    assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), -2);
+                });
         ledgerConf = pulsar.getBrokerService().getManagedLedgerConfig(topicName).get();
         offloader = new MockLedgerOffloader(OffloadPoliciesImpl.create("S3", "", "", "",
                 null, null,
@@ -1534,7 +1516,10 @@ public class NamespacesTest extends MockedPulsarServiceBaseTest {
         // set back to -1 and fall back to default
         admin.namespaces().setOffloadThreshold(namespace, -1);
         admin.namespaces().setOffloadThresholdInSeconds(namespace, -1);
-        assertEquals(-1, admin.namespaces().getOffloadThreshold(namespace));
+        Awaitility.await().untilAsserted(() -> {
+                    assertEquals(admin.namespaces().getOffloadThreshold(namespace), -1);
+                    assertEquals(admin.namespaces().getOffloadThresholdInSeconds(namespace), -1);
+                });
         ledgerConf = pulsar.getBrokerService().getManagedLedgerConfig(topicName).get();
         offloader = new MockLedgerOffloader(OffloadPoliciesImpl.create("S3", "", "", "",
                 null, null,

